pkg/bloomgateway/bloomgateway.go

```diff
@@ -238,31 +238,50 @@ func (g *Gateway) FilterChunkRefs(ctx context.Context, req *logproto.FilterChunk
 		}, nil
 	}
 
+	blocks := make([]bloomshipper.BlockRef, 0, len(req.Blocks))
+	for _, key := range req.Blocks {
+		block, err := bloomshipper.BlockRefFromKey(key)
+		if err != nil {
+			stats.Status = labelFailure
+			return nil, errors.New("could not parse block key")
+		}
+		blocks = append(blocks, block)
+	}
+
+	// Shortcut if request does not contain blocks
+	// TOOD(chaudum): Comment out when blocks are resolved on index gateways
```

Comment:
I would have removed that after the weekly release, but you are right, since we don't run this in ops yet, we don't need to care that much about compatibility.